pkg/zk/mod/mod.go

```diff
@@ -206,15 +209,14 @@ func (p *Proof) Verify(hash *hash.Hash, public Public) bool {
 	// get [yᵢ] <- ℤₙ
 	ys := challenge(hash, n, p.W)
 
-	var lhs, rhs big.Int
-	var z, x *big.Int
 	four := big.NewInt(4)
-	for i := 0; i < params.StatParam; i++ {
+	verifications := pl.Parallelize(params.StatParam, func(i int) interface{} {
```

Comment:
Maybe make the function stop early if any of the verifications fail?
This wouldn't matter in practice, since the loop runs for the full number of iterations.

Reply:
You could add functionality to the pool for that, yeah.

My reasoning is that 99% of the time you're going to have to check everything because 99% of the time parties are honest. So this would speedup the case where the proof fails, but that's not common.